app/upgrades/v23/upgrades.go

```diff
@@ -0,0 +1,126 @@
+package v23
+
+import (
+	"fmt"
+
+	icqtypes "github.com/cosmos/ibc-apps/modules/async-icq/v7/types"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	"github.com/CosmosContracts/juno/v23/app/keepers"
+	"github.com/CosmosContracts/juno/v23/app/upgrades"
+)
+
+type IndividualAccount struct {
+	Owner   string
+	Address string
+}
+
+// Core1VestingAccounts https://daodao.zone/dao/juno1j6glql3xmrcnga0gytecsucq3kd88jexxamxg3yn2xnqhunyvflqr7lxx3/members
+// we are including only lobo, dimi and jake because the other ones do not agree on giving up their vesting
+var Core1VestingAccounts = []IndividualAccount{
+	{
+		Owner:   "dimi",
+		Address: "juno1s33zct2zhhaf60x4a90cpe9yquw99jj0zen8pt",
+	},
+	{
+		Owner:   "jake",
+		Address: "juno18qw9ydpewh405w4lvmuhlg9gtaep79vy2gmtr2",
+	},
+	{
+		Owner:   "wolf",
+		Address: "juno1a8u47ggy964tv9trjxfjcldutau5ls705djqyu",
+	},
+}
+
+func CreateV23UpgradeHandler(
+	mm *module.Manager,
+	cfg module.Configurator,
+	keepers *keepers.AppKeepers,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		logger := ctx.Logger().With("upgrade", UpgradeName)
+
+		nativeDenom := upgrades.GetChainsDenomToken(ctx.ChainID())
+		logger.Info(fmt.Sprintf("With native denom %s", nativeDenom))
+
+		// migrate ICQ params
+		for _, subspace := range keepers.ParamsKeeper.GetSubspaces() {
+			subspace := subspace
+
+			var keyTable paramstypes.KeyTable
+			if subspace.Name() == icqtypes.ModuleName {
+				keyTable = icqtypes.ParamKeyTable()
+			} else {
+				continue
+			}
+
+			if !subspace.HasKeyTable() {
+				subspace.WithKeyTable(keyTable)
+			}
+		}
+
+		// Migrate Tendermint consensus parameters from x/params module to a deprecated x/consensus module.
+		// The old params module is required to still be imported in your app.go in order to handle this migration.
+		baseAppLegacySS := keepers.ParamsKeeper.Subspace(baseapp.Paramspace).WithKeyTable(paramstypes.ConsensusParamsKeyTable())
+		baseapp.MigrateParams(ctx, baseAppLegacySS, &keepers.ConsensusParamsKeeper)
```

Comment:
I don't think we need to migrate consensus params here since it already been done in the [v16 upgrade](https://github.com/CosmosContracts/juno/blob/main/app/upgrades/v16/upgrades.go#L113).

Reply:
good catch!

Thanks